src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs

```diff
@@ -987,18 +987,28 @@ private static string EnsurePackageExtension(string packagePath, bool isSnupkg)
         }
 
         /// <summary>
-        /// This method checks of a given list of package paths exist on disk.
+        /// If there isn't at least one Path specified, throw that no file paths were resolved for this Package.
         /// </summary>
-        /// <param name="packagePath">A package path to be used while creating logs.</param>
-        /// <param name="matchingPackagePaths">A list of matching package paths that need to be checked.</param>
+        /// <param name="packagePath">The package path the user originally provided for use while creating logs.</param>
+        /// <param name="matchingPackagePaths">A list of matching package paths that were previously resolved.</param>
         public static void EnsurePackageFileExists(string packagePath, IEnumerable<string> matchingPackagePaths)
         {
-            if (!matchingPackagePaths.Any())
+            if (!ResolvedAnyPackagePath(matchingPackagePaths))
             {
+                ErrorFileNotFound(packagePath);
+            }
+        }
+
+        public static void ErrorFileNotFound(string packagePath)
+        {
                 throw new ArgumentException(string.Format(CultureInfo.CurrentCulture,
                     Strings.UnableToFindFile,
                     packagePath));
-            }
+        }
+
+        public static bool ResolvedAnyPackagePath(IEnumerable<string> matchingPackagePaths)
+        {
+            return matchingPackagePaths != null && matchingPackagePaths.Any();
         }
```

Comment:
Lots of topics here, but the most interesting is what I've already discussed with @nkolev92: if we are going to have public utility classes, then they should be complete, or we should strongly evaluate their existence.

You're suggesting I change the type of the Exception that our public API has been throwing since this utility was created. I'm not sure if there's significant impact to doing this. Can you provide any knowledge you have on impact?

Finally, you're suggesting that the term "resolve" be renamed due to confusion. I can refactor this entire Utility to call it something else. Again, I'll ask for your input on the impact that renaming will have. 

Any of the above could warrant a team discussion about the direction we want to take with future changes as well.

Reply:
> @nkolev92: if we are going to have public utility classes, then they should be complete, or we should strongly evaluate their existence.

I think you're clinging to straws here. I brought up similar feedback as Andy about the extra public methods.
I fully agree with Andy, and I don't like the addition of simple utility methods with no real business logic.

Reply:
I'm adding this comment to help explain my thought process, so I hope it doesn't come off as naggy or overbearing.

> if we are going to have public utility classes, then they should be complete

I don't see how helper functions to throw an exception or check if a list is empty makes "local folder utility" more complete. Neither of those methods seem related to dealing with local folders in my opinion. Now, in another comment you did write that you wanted the method name to explain intent, so I now I can see how it fits in this utility class for that reason, but that idea didn't come naturally to me, it had to be pointed out, which makes me feel like it could be improved. Anyway, consider this:

```csharp
var nupkgsToPush = LocalFolderUtility.ResolvePackageFromPath(packagePath, isSnupkg: false); 
if (nupkgsToPush.Count == 0) {  /* error case */ }
```
Replace `Count == 0` with `!.Any()` or whatever, I find this much easier to read and understand. Partly because my working/thinking style is very detail oriented. Partly because almost my entire career I've worked on software that was 10+ years old before I joined the team, and many comments and method names were misleading because they weren't changed when code was refactored or extended. Also, most methods are non-trivial, so when I see a method being called in an if statement, even if I trust the name, I expect it's because the check is complex, which increases my mental-load, even if it turns out that it's trivial. These are the reasons I don't like helper methods that are just one liners.

> You're suggesting I change the type of the Exception that our public API has been throwing since this utility was created.

This is a perfect example of the misunderstandings we have :)  I was suggesting that the name of the method was misleading. Frankly I have trouble understanding how my previous message could be interpreted as asking for the exception being thrown to change. I agree that changing the exception type has risk and should be avoided. But `FileNotFoundException` is a well known class, so method names shouldn't suggest that it's being thrown when a different exception type is actually being thrown, even if the reason is because a file wasn't found. Naming things based on intent sounds like a good idea, but when it introduces confusion because of common, well-known technical concepts or details, then the name has hurt understanding rather than helped.

Reply:
Removed both helper methods. Did not change the existing Exception being thrown.